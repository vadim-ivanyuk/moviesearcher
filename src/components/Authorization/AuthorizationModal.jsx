import React from "react";
import { Modal, ModalBody } from "reactstrap";
import PropTypes from "prop-types";
import { AuthorizationForm } from "./AuthorizationForm";

export const AuthorizationModal = (props) => {
  const { modal, toggle } = props;

  return (
    <Modal isOpen={modal} toggle={toggle} className="authorization-form-modal">
      <ModalBody>
        <AuthorizationForm />
      </ModalBody>
    </Modal>
  );
};

AuthorizationModal.defaultProps = {
  modal: false,
};

AuthorizationModal.propTypes = {
  modal: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
};
