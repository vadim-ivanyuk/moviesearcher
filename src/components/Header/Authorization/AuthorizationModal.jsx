import React from "react";
import { Modal, ModalBody } from "reactstrap";
import AuthorizationForm from "./Form/AuthorizationForm";

export const AuthorizationModal = (props) => {
  const { modal, toggle } = props;

  return (
    <Modal isOpen={modal} toggle={toggle} className="authorization-form-modal">
      <ModalBody>
        <AuthorizationForm />
      </ModalBody>
    </Modal>
  );
};
