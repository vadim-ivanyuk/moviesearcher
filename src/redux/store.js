import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./rootReducer";
import { ON_CHANGE_FILTERS } from "./filters/filters.types";

let currentPage;

const onChangePageMiddleware = ({ dispatch, getState }) => (next) => (
  action
) => {
  currentPage = getState().filters.page;
  if (
    action.type === ON_CHANGE_FILTERS &&
    action.payload.name !== "page" &&
    currentPage !== 1
  ) {
    dispatch({
      type: ON_CHANGE_FILTERS,
      payload: {
        name: "page",
        value: 1,
      },
    });
  }

  return next(action);
};

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, onChangePageMiddleware))
);

export default store;
