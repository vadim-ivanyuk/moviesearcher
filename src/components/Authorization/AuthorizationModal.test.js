import React from "react";
import { configure, shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "../../redux/store";
import { AuthorizationModal } from "./AuthorizationModal";

configure({ adapter: new Adapter() });

describe("Authorization modal", () => {
  it("snap Authorization modal", () => {
    const authorizationModal = shallow(
      <Provider store={store}>
        <AuthorizationModal modal={false} toggleModal={jest.fn()} />
      </Provider>
    );
    expect(shallowToJson(authorizationModal)).toMatchSnapshot();
  });
});
