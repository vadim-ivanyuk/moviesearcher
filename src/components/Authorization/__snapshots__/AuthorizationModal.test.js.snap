// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization modal snap Authorization modal 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <AuthorizationModal
    modal={false}
    toggleModal={[MockFunction]}
  />
</ContextProvider>
`;
